editors/code/src/tasks.ts

```diff
@@ -2,22 +2,20 @@ import * as vscode from "vscode";
 import * as toolchain from "./toolchain";
 import type { Config } from "./config";
 import { log } from "./util";
-import { unwrapUndefinable } from "./undefinable";
 
 // This ends up as the `type` key in tasks.json. RLS also uses `cargo` and
 // our configuration should be compatible with it so use the same key.
 export const TASK_TYPE = "cargo";
 export const TASK_SOURCE = "rust";
 
-export interface CargoTaskDefinition extends vscode.TaskDefinition {
-    command?: string;
-    args?: string[];
+export interface RustTargetDefinition extends vscode.TaskDefinition {
+    program: string;
+    args: string[];
     cwd?: string;
     env?: { [key: string]: string };
-    overrideCargo?: string;
 }
 
-class CargoTaskProvider implements vscode.TaskProvider {
+class RustTaskProvider implements vscode.TaskProvider {
```

Comment:
nit: I'm not sure there will be any *other* kind of Task run by rust-analyzer, so I wouldn't qualify with `Rust`, but I also don't feel too strongly.

Reply:
I think it needs *some* name, `class TaskProvider implements vscode.TaskProvider` would be confusing. I don't feel strongly about the name though. Do you have any suggestions? `RustAnalyzerTaskProvider` is maybe a little long.